src/pygeoprocessing/geoprocessing.py

```diff
@@ -3788,3 +3584,352 @@ def raster_to_numpy_array(raster_path, band_id=1):
+            raster_info['bounding_box'],
+            raster_info['projection_wkt'],
+            target_raster_info['projection_wkt'])
+
+        try:
+            # merge the bounding boxes only to see if they don't intersect
+            _ = merge_bounding_box_list(
+                [projected_raster_bounding_box,
+                 target_raster_info['bounding_box']], 'intersection')
+        except ValueError:
+            LOGGER.warning(
+                f'the raster at "{raster_path}"" does not intersect the '
+                f'stitch raster at "{target_stitch_raster_path_band[0]}", '
+                'skipping...')
+            continue
+
+        # use this to determine if we need to warp and delete if we did at
+        # the end
+        if (raster_info['projection_wkt'] ==
+            target_raster_info['projection_wkt'] and
+            raster_info['pixel_size'] ==
+                target_raster_info['pixel_size']):
+            warped_raster = False
+            base_stitch_raster_path = raster_path
+        else:
+            workspace_dir = tempfile.mkdtemp(
+                dir=os.path.dirname(target_stitch_raster_path_band[0]),
+                prefix='stitch_rasters_workspace')
+            base_stitch_raster_path = os.path.join(
+                workspace_dir, os.path.basename(raster_path))
+            warp_raster(
+                raster_path, target_raster_info['pixel_size'],
+                base_stitch_raster_path, resample_method,
+                target_projection_wkt=target_raster_info['projection_wkt'],
+                working_dir=workspace_dir,
+                osr_axis_mapping_strategy=osr_axis_mapping_strategy)
+            warped_raster = True
+
+        if warped_raster and area_weight_m2_to_wgs84:
+            # determine base area per pixel currently and area per pixel
+            # once it is projected to wgs84 pixel sizes
+            base_pixel_area_m2 = abs(numpy.prod(raster_info['pixel_size']))
+            base_stitch_raster_info = get_raster_info(
+                base_stitch_raster_path)
+            lat_min, lat_max = [
+                base_stitch_raster_info['bounding_box'][_]
+                for _ in [1, 3]]
+            n_rows = base_stitch_raster_info['raster_size'][1]
+            # this column is a longitude invariant latitude variant pixel
+            # area for scaling area dependent values
+            m2_area_per_lat = _create_latitude_m2_area_column(
+                lat_min, lat_max, n_rows)
+
+            def _mult_op(base_array, base_nodata, scale):
+                """Scale non-nodata by scale."""
+                result = numpy.empty(base_array.shape)
+                result[:] = base_array
+                if base_nodata is not None:
+                    valid_mask = ~numpy.isclose(base_array, base_nodata)
+                else:
+                    valid_mask = numpy.ones(
+                        base_array.shape, dtype=numpy.bool)
```

Comment:
We [also learned fairly recently](https://github.com/natcap/invest/pull/269#issuecomment-679471260) that `slice(None)` is the equivalent of `[:]`, and offers order-of-magnitude faster indexing.  Would it be worth using?

Reply:
Where should i use `slice(None)`? Instead of `numpy.ones`?

Reply:
I think you meant on line 3752 right? Changed in 073dfdc

Reply:
Actually, I was thinking of cases where we construct a boolean array where we select all pixels (`valid_mask = numpy.ones(base_array.shape, dtype=numpy.bool)`).  The point of `slice(None)` is that you can assign it to a variable, which is great when we're conditionally constructing a mask.

```
a = slice(None)  # Works great, equivalent of array[:], so you could now say array[a] == array[:]
b = :  # This is a syntax error, of course
```

Reply:
Okay I thought I understood, but when I tried what I thought you were saying I'm not seeing a consistent speedup. Can you check? Or maybe this is already taking too long and isn't really about this PR... Here's a gist of what I tried, I was trying to make a real world scenario with rando nodata values and some kind of oepration etc. I only time the parts where the functions are called, not the setup: https://gist.github.com/richpsharp/94a65b0b01c6fffcb47ec00c999f7938

Running a few times I see this:

```
(py38) Rich@DESKTOP-ETOEL2F D:\repositories
> python timeit.py
D: 256
op_time: 0.1299595832824707
noneslice_op_time: 0.13809561729431152
noneslice performance increase: 94.11

(py38) Rich@DESKTOP-ETOEL2F D:\repositories
> python timeit.py
D: 256
op_time: 0.1300041675567627
noneslice_op_time: 0.14000415802001953
noneslice performance increase: 92.86

(py38) Rich@DESKTOP-ETOEL2F D:\repositories
> python timeit.py
D: 256
op_time: 1.1344609260559082
noneslice_op_time: 1.1809608936309814
noneslice performance increase: 96.06

(py38) Rich@DESKTOP-ETOEL2F D:\repositories
> python timeit.py
D: 256
op_time: 1.1216988563537598
noneslice_op_time: 1.1957592964172363
noneslice performance increase: 93.81

(py38) Rich@DESKTOP-ETOEL2F D:\repositories
> python timeit.py
D: 256
op_time: 1.1153075695037842
noneslice_op_time: 1.1903018951416016
noneslice performance increase: 93.70

(py38) Rich@DESKTOP-ETOEL2F D:\repositories
> python timeit.py
D: 256
op_time: 1.1111719608306885
noneslice_op_time: 1.1871743202209473
noneslice performance increase: 93.60
```

Basically I see the `slice(None)` is always a liiiitle bit slower. That's what I'd kinda expect since I'm turning one function call into two and I'd sort of hope the intializers that numpy provides are actually tuned to be used for initialization. Can you take a look at that? And/or maybe we can move this to slack instead of this PR assuming you don't need me to do anything else here?

Reply:
Ach, sorry ... The `slice(None)` advantage here is only useful in a _very_ specific case, and I agree, no need to pursue it further here.  I'll follow up about this in slack a little later.